### PR TITLE
Merge `develop` into `main`

### PR DESCRIPTION
**Pull request by falkben**
Combination of 3 features

- docs: https://github.com/spacetelescope/DRESSCode/pull/25
- github actions for docker build/push and test of pipeline: https://github.com/spacetelescope/DRESSCode/pull/28 and https://github.com/spacetelescope/DRESSCode/pull/21
- spacetelescope style guide: https://github.com/spacetelescope/DRESSCode/pull/24

### CODE REVIEW

**Review by mdecleir — COMMENTED** [review]
Looks great!
Here are some comments/questions:

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+- We pledge to welcome questions and answer them respectfully, paying particular attention to those new to the community. We pledge to provide respectful criticisms and feedback in forums, especially in discussion threads resulting from code contributions.
+
+- We pledge to be conscientious of the perceptions of the wider community and to respond to criticism respectfully. We will strive to model behaviors that encourage productive debate and disagreement, both within our community and where we are criticized. We will treat those outside our community with the same respect as people within our community.
+
+- We pledge to help the entire community follow the code of conduct, and to not remain silent when we see violations of the code of conduct. We will take action when members of our community violate this code such as such as contacting conduct@stsci.edu (all emails sent to this address will be treated with the strictest confidence) or talking privately with the person.
```
[the remark this answers is not in the document]
Comment (falkben) [reply CODE_OF_CONDUCT.md:17]: ah, nice, do you want to submit a PR to the style guide: https://github.com/spacetelescope/style-guides

File: .github/workflows/test-pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          cp -r ~/Raw_data ~/dresscode-data/NGC0628/Raw_data
+          chmod -R a=rwx ~/dresscode-data
+          docker run --user root --rm -t -v ~/dresscode-data:/data dresscodeswift/dresscode /bin/bash /opt/dresscode/tests/test_pipeline.bash /opt/dresscode /data
+
+      - name: Save pipeline output
+        uses: actions/upload-artifact@v2
+        with:
+          name: final-fits
+          path: |
+            ~/dresscode-data/NGC0628/working_dir/*final*.fits
```
[the remark this answers is not in the document]
Comment (falkben) [reply .github/workflows/test-pipeline.yml:47]: it's all saved in the run. you can see it if you click the individual steps of the github action run.

If you click "Run the pipeline" here you should see it all: https://github.com/spacetelescope/DRESSCode/runs/2806375797

Or you can alternatively see the entire "raw logs" as well: https://pipelines.actions.githubusercontent.com/DZYC2omuZ7CATA2T87xcqCvwXFOkwfKcTnpqlAddLV6PvsVv3W/_apis/pipelines/1/runs/76/signedlogcontent/3?urlExpires=2021-06-14T21%3A07%3A07.3011225Z&urlSigningMethod=HMACV1&urlSignature=OEH4SgwV%2BSrQ3TgBIUVF%2FrZMfsIZvgNzfEem%2FGIZYk8%3D

File: .github/workflows/test-pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Push the dependency Docker image
+        if: github.ref == 'refs/heads/main'
+        run: docker push dresscodeswift/heasoft-caldb-wcstools:latest
+
+      - name: Push the DRESSCode Docker image
+        if: github.ref == 'refs/heads/main'
+        run: docker push dresscodeswift/dresscode:latest
```
[the remark this answers is not in the document]
Comment (falkben) [reply .github/workflows/test-pipeline.yml:62]: if it gets this far, it's gotten through the "test pipeline" step without error

File: Docker/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+|heasoft w/ caldb & wcstools|[Docker/heasoft-caldb-wcstools.dockerfile](/Docker/heasoft-caldb-wcstools.dockerfile)|[heasoft-caldb-wcstools ![Docker Pulls](https://img.shields.io/docker/pulls/dresscodeswift/heasoft-caldb-wcstools) ![Docker Image Size (latest by date)](https://img.shields.io/docker/image-size/dresscodeswift/heasoft-caldb-wcstools)](https://hub.docker.com/repository/docker/dresscodeswift/heasoft-caldb-wcstools)|adds caldb files and wcstools|
+|dresscode|[Docker/dockerfile](/Docker/dockerfile)|[dresscode ![Docker Pulls](https://img.shields.io/docker/pulls/dresscodeswift/dresscode) ![Docker Image Size (latest by date)](https://img.shields.io/docker/image-size/dresscodeswift/dresscode)](https://hub.docker.com/repository/docker/dresscodeswift/dresscode)|adds DRESSCode pipeline|
```
[the remark this answers is not in the document]
Comment (falkben) [reply Docker/README.md:15]: yes, it contains the output of that dockerfile. you shouldn't need both as an end user.

File: .github/workflows/test-pipeline.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mdecleir) [reply .github/workflows/test-pipeline.yml:47]: That's great, thank you!

(That second URL seems to be broken though.)

File: CODE_OF_CONDUCT.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+As members of the community,
+
+- We pledge to treat all people with respect and provide a harassment- and bullying-free environment, regardless of sex, sexual orientation and/or gender identity, disability, physical appearance, body size, race, nationality, ethnicity, and religion. In particular, sexual language and imagery, sexist, racist, or otherwise exclusionary jokes are not appropriate.
+
+- We pledge to respect the work of others by recognizing acknowledgment/citation requests of original authors. As authors, we pledge to be explicit about how we want our own work to be cited or acknowledged.
```
[the remark this answers is not in the document]
Comment (falkben) [reply CODE_OF_CONDUCT.md:9]: Made an issue for this and assigned you :) https://github.com/spacetelescope/DRESSCode/issues/35

**Review by mdecleir — APPROVED** [review]
These changes look good!

**Pull request merged**